src/main/java/keycontacts/logic/commands/ScheduleCommand.java

```diff
@@ -66,7 +67,18 @@ public CommandResult execute(Model model) throws CommandException {
         if (studentToUpdate.equals(updatedStudent)) {
             throw new CommandException(MESSAGE_LESSON_UNCHANGED);
         }
-        model.setStudent(studentToUpdate, updatedStudent);
+
+        Student studentToUpdateWithoutRegularLesson = studentToUpdate.withoutRegularLesson();
+        model.setStudent(studentToUpdate, studentToUpdateWithoutRegularLesson);
+
+        ClashResult clashResult = model.checkClashingLesson(regularLesson);
+        if (clashResult.hasClash()) {
+            model.setStudent(studentToUpdateWithoutRegularLesson, studentToUpdate); // revert change if clash
+            throw new CommandException(String.format(MESSAGE_CLASHING_LESSON,
+                    Messages.format(clashResult.getClashingLesson())));
+        }
+
+        model.setStudent(studentToUpdateWithoutRegularLesson, updatedStudent);
```

Comment:
I will likely update this implementation in the group feature, since multiple students will be able to share the same regular lesson. No change needed here for now